.travis.yml

```diff
@@ -45,18 +54,21 @@ before_install: |
     pyenv install $PYTHON 
     pyenv virtualenv $PYTHON MYVERSION
     source ~/.pyenv/versions/MYVERSION/bin/activate
+    pyenv global $PYTHON
     python --version 
     python -c 'import sys;print(sys.version)' 
     python -c 'import sys;print(sys.executable)' 
   fi
 install:
 # we have this here so we can see where python is installed and hardcode in our tests
 # else when running npm test, the python version picked is completely different :(
+# python-dev is to ensure jupyter installs properly
+  - sudo apt-get install python-dev
```

Comment:
Can you not specify `python-dev` under `packages` so you can drop the `sudo` requirement? I'm actually surprised it isn't included by default.

Reply:
For some reason this is required, else installing jupyter falls over.

Reply:
Hrm, that's unfortunate/weird. Another reason to eventually rip that code out! 😉

Reply:
Yes agreed.
However I would like to remove Jupyter tests and all of the related code/settings in a separate PR. I'd like to track that separately. what do you think?
I.e. leave this for now, and it will get removed in another PR.